model/src/main/java/org/cbioportal/model/GeneticProfile.java

```diff
@@ -20,7 +20,16 @@
         PHOSPHORYLATION,
         PROTEIN_LEVEL,
         PROTEIN_ARRAY_PROTEIN_LEVEL,
-        PROTEIN_ARRAY_PHOSPHORYLATION;
+        PROTEIN_ARRAY_PHOSPHORYLATION, 
+        GENESET_SCORE;
+    }
+    
+    public static class DataType {
```

Comment:
You can take a look at SampleType enum in Sample model, it doesn't have `-` but it has white space in it, can be solved the same way I think. You will need a MyBatis type handler like `org.cbioportal.persistence.mybatis.typehandler.SampleTypeTypeHandler` for conversion. And `super.enable(SerializationFeature.WRITE_ENUMS_USING_TO_STRING);` in CustomObjectMapper will use the toString method of enum, so in JSON response you will get `P-VALUE` instead of `P_VALUE`. (You need to return the value from toString method in your enum, just like SampleType enum).